lib/matplotlib/figure.py

```diff
@@ -2031,7 +2026,9 @@ def make_active(event):
 
     def add_axobserver(self, func):
         """Whenever the axes state change, ``func(self)`` will be called."""
-        self._axobservers.append(func)
+        # Connect a wrapper lambda and not func itself, to avoid it being
+        # weakref-collected.
+        self._axobservers.connect("_on_axes_change", lambda arg: func(arg))
```

Comment:
```suggestion
        self._axobservers.connect("_axes_change_event", lambda arg: func(arg))
```

in analogy to e.g. `canvas.mpl_connect('key_press_event', self.on_key_press)`. The event is called `*_event`, the callback `on_*`.

_

Reply:
sure, done